### DISCUSSION TITLE
Cannot compile

### DISCUSSION THREAD

**Discussion by alfredolo72** [bertmelis/VitoWiFi#100] [General]
software : Vs code + platformio
hardware : esp32dev

I'm trying to compile version 3 with an esp32dev but I'm encountering errors. First of all, this is my manifest :

[common]
build_flags =
  -D DEBUG_VITOWIFI=1
  -D CORE_DEBUG_LEVEL=ARDUHAL_LOG_LEVEL_VERBOSE
  -Wall
  -Wextra
  -Wpedantic
  -std=c++11

[env:esp32dev]
platform = espressif32
framework = arduino
board = esp32dev
lib_deps =
  VitoWiFi
  plerup/espsoftwareserial
build_flags =
  ${common.build_flags}
  -lgcov
  --coverage
  -D VW_START_PAYLOAD_LENGTH=10
extra_scripts = test_coverage.py

During the compilation, it was giving me an error and I changed "VitoWiFi.hpp" from : 
  void _readDatapoint(IDatapoint* dp, void* arg);
  void _writeDatapoint(IDatapoint* dp, DPValue value, void* arg);
to : 
  bool _readDatapoint(IDatapoint* dp, void* arg);
  bool _writeDatapoint(IDatapoint* dp, DPValue value, void* arg);

Now after almost 2 mins of compilation i get this error : 
collect2.exe: error: ld returned 1 exit status
*** [.pio\build\esp32dev\firmware.elf] Error 1

How can i fix it ?

**Comment by bertmelis** [comment]
try

```
lib_deps =
  plerup/espsoftwareserial
  https://github.com/bertmelis/VitoWiFi.git#v3
```

(specifying softwareserial isn't really needed I think Platformio should pull it automatically as dependency.)

v3 isn't released yet and also isn't the default branch on Github. So you have to explicitly set it that way. Also, check the readme and the examples in this branch. Things have changed a bit.
